### PR TITLE
Keep title on preview project page from overflowing or otherwise messing up the layout

### PR DESCRIPTION
**Pull request by LiFaytheGoblin**
### Resolves:

Part of #1986 

### Changes:

* added a title attribute to project title so you can see the whole title even when it is not completely displayed on the website
* made it so long titles don't get displayed completely, see comments in scss file

### Test Coverage:
npm test passed
Tested in MacOS Chrome & Safari, looks good for screen with min-width 640px

screen shot 2018-08-23 at 2 32 50 pm

screen shot 2018-08-23 at 2 32 09 pm

### CODE REVIEW

File: src/views/preview/preview.scss [hunk 1]
```diff
@@ -32,13 +32,23 @@ $stage-width: 480px;
...
+            show the full title if this is too long */
+            overflow: hidden;
+            text-overflow: ellipsis;
+            white-space: nowrap;
+        }
     }
 
     .project-header {
         margin-right: 2rem;
+        width: 65%; /* Setting a width to the header will keep it from flowing over the usual content of the page */
```
Comment (rschamp) [inline src/views/preview/preview.scss]: Where does `65%` come from? I would expect this to be 100% and constrained by the parent component with margins or flex-basis or something like that?

Comment (LiFaytheGoblin) [reply src/views/preview/preview.scss]: I have tried for two hours now to get it to work without setting the width this way but it doesn't. Flex basis seems to require a width also. Changing settings for the button doesn't change the width of that input field either. I think putting the width here is the easiest and for what I have tested safest thing we can do.

Comment (LiFaytheGoblin) [reply src/views/preview/preview.scss]: ACTUALLY this is so interesting that I am now reading more about flex box in general and I might have found a solution. Will investigate further. Hint: min-width 0!

Comment (LiFaytheGoblin) [reply src/views/preview/preview.scss]: OMG RAY IT WORKS!

**Comment by LiFaytheGoblin** [comment]
Found a nice way to do this thanks to https://hackernoon.com/11-things-i-learned-reading-the-flexbox-spec-5f0c799c776b and the hint about min-width

**Comment by LiFaytheGoblin** [comment]
@rschamp Thanks ;D Should I merge?

**Pull request merged**